modules/misc/qt.nix

```diff
@@ -147,37 +187,31 @@ in {
       '';
     }];
 
-    qt.style.package = mkIf (cfg.style.name != null)
-      (mkDefault (stylePackages.${toLower cfg.style.name} or null));
-
-    home.sessionVariables = envVars;
+    qt.style.package = lib.mkIf (cfg.style.name != null)
+      (lib.mkDefault (stylePackages.${lib.toLower cfg.style.name} or null));
+
+    home = {
+      sessionVariables = envVars;
+      # home.sessionVariables does not support setting the same environment
+      # variable to different values.
+      # Since some other modules may set the QT_PLUGIN_PATH or QML2_IMPORT_PATH
+      # to their own value, e.g.: fcitx5, we avoid conflicts by setting
+      # the values in home.sessionVariablesExtra instead.
+      sessionVariablesExtra = ''
+        export QT_PLUGIN_PATH=${envVarsExtra.QT_PLUGIN_PATH}
+        export QML2_IMPORT_PATH=${envVarsExtra.QML2_IMPORT_PATH}
+      '';
+    };
 
     # Apply theming also to apps started by systemd.
-    systemd.user.sessionVariables = envVars;
-
-    home.packages = (if cfg.platformTheme == "gnome" then
-      [ pkgs.qgnomeplatform ]
-    else if cfg.platformTheme == "qtct" then [
-      pkgs.libsForQt5.qt5ct
-      pkgs.qt6Packages.qt6ct
-    ] else if cfg.platformTheme == "kde" then [
-      pkgs.libsForQt5.plasma-integration
-      pkgs.libsForQt5.systemsettings
-    ] else
-      [ pkgs.libsForQt5.qtstyleplugins ])
+    systemd.user.sessionVariables = envVars // envVarsExtra;
+
+    home.packages = (platformPackages.${cfg.platformTheme} or [ ])
       ++ lib.optionals (cfg.style.package != null)
       (lib.toList cfg.style.package);
 
-    xsession.importedVariables = [ "QT_QPA_PLATFORMTHEME" ]
+    xsession.importedVariables =
+      [ "QT_QPA_PLATFORMTHEME" "QT_PLUGIN_PATH" "QML2_IMPORT_PATH" ]
```

Comment:
Not familiar with the xsession stuff (I use wayland). Do we want `QT_QPA_PLATFORMTHEME` here if we're not setting it?

Reply:
I think there is no harm setting it if it is not set, but not sure either.

Reply:
Oddly enough that option doesn't show up in the docs, but as far as I can tell it just passes the variable name to `systemctl --user import-environment` in ~/.xsession.

Looking at [the man page](https://www.freedesktop.org/software/systemd/man/latest/systemctl.html#%0A%20%20%20%20%20%20%20%20%20%20%20%20import-environment%0A%20%20%20%20%20%20%20%20%20%20%20%20VARIABLE%E2%80%A6%0A%20%20%20%20%20%20%20%20%20%20):
>Import all, one or more environment variables set on the client into the systemd manager environment block. If a list of environment variable names is passed, client-side values are then imported into the manager's environment block. If any names are not valid environment variable names or have invalid values according to the rules described above, an error is raised.

I'm not entirely sure how to interpret this, what constitutes an invalid variable name?

Empirically:
```
$ systemctl --user import-environment arstarstarstarst
Environment variable $arstarstarstarst not set, ignoring.
```
It just prints a warning and exits with status code 0, so I think it doesn't strictly matter, but I would still gate this to suppress the warning.

Reply:
Done.